spec/requests/posts/index_spec.rb

```diff
@@ -1,24 +1,58 @@
 require 'rails_helper'
 require 'capybara/rspec'
 
-RSpec.describe 'Post index page', type: :request do
-  it 'displays the posts and their details' do
-    # Create a test user
-    user = User.create(name: 'John Doe', photo: 'valid_photo_url')
+RSpec.describe 'User profile page', type: :feature do
+  before do
+    @user = User.create(name: 'Thomas Heflord',
+                        photo: 'https://media.istockphoto.com/id/1289461328/photo/portrait-of-a-handsome-black-man.jpg?s=612x612&w=0&k=20&c=y_mzB0Tbe5LErNy6pqfY7sz2HiDT7fOAUCwupN3-Bg4=',
+                        posts_counter: 5)
+    @posts = [
+      Post.create(user: @user, title: 'Morality',
+                  text: 'Strive for Personal Growth: Embrace personal growth and strive to become the best',
+                  comments: []),
+      Post.create(user: @user, title: 'Post 2 ',
+                  text: 'Sample text',
+                  comments: []),
+      Post.create(user: @user, title: 'Post 3',
+                  text: 'Practice integrity and honesty in your words and actions. Be true to your principles and',
+                  comments: []),
+      Post.create(user: @user, title: 'Post 4',
+                  text: 'Sample text',
+                  comments: []),
+      Post.create(user: @user, title: 'Post 5',
+                  text: 'Sample text',
+                  comments: [])
+    ]
+  end
+  it 'displays the user profile information and posts' do
+    visit user_path(@user)
+
+    expect(page).to have_content(@user.name)
+    expect(page).to have_selector('img[src="https://media.istockphoto.com/id/1289461328/photo/portrait-of-a-handsome-black-man.jpg?s=612x612&w=0&k=20&c=y_mzB0Tbe5LErNy6pqfY7sz2HiDT7fOAUCwupN3-Bg4="]')
+    expect(page).to have_content('Number of posts: 5')
 
-    # Create some test posts associated with the user
-    post1 = Post.create(user:, title: 'Post 1', text: 'Content of Post 1')
-    post2 = Post.create(user:, title: 'Post 2', text: 'Content of Post 2')
+    @user.posts.each do |post|
+      expect(page).to have_selector('.post-container', text: post.text)
+      within('.post-container', text: post.text) do
+        expect(page).to have_content(post.text)
+        expect(page).to have_content(post.text)

```

Comment:
#### Good job adding most of the tests @TayyabKM, we still miss one more test.  
- [ ] Tests that assert you can see a post's title.
- [Optional] for the number of posts the user has written. consider updating the test to use a customer method instead of a hard-coded number. Considering the number of posts is going to increase/decrease its better to get this data from the user itself. 
> Something like...
```ruby 
expect(page).to have_content("Number of posts: #{@user.posts.count}")
```